nj2k/src/org/jetbrains/kotlin/nj2k/conversions/DefaultArgumentsConversion.kt

```diff
@@ -125,6 +133,52 @@ class DefaultArgumentsConversion(context: NewJ2kConverterContext) : RecursiveApp
 
     }
 
+    private fun JKExpression.addThisReceiverIfNeeded(
+        parameters: List<JKParameter>,
+        declarations: List<JKDeclaration>
+    ): JKExpression? = when (this) {
+        is JKFieldAccessExpression, is JKCallExpression -> {
+            if (parameters.any { it.name.value == identifier?.name } && declarations.any { it == identifier?.target }) {
+                parent?.also { this.detach(it) }
```

Comment:
Please, don't use detach directly. The `element::child.detached()` syntax is better to be used as it also check parent-child relation consistency. I will deprecate this method to avoid confusions